Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Tooltips/ToolTip.cs

```diff
@@ -0,0 +1,540 @@
+        {
+            get
+            {
+                return contentParent.transform.TransformPoint(localAttachPoint) + attachPointOffset;
+            }
+            set
+            {
+                // apply the difference to the offset
+                attachPointOffset = value - contentParent.transform.TransformPoint(localAttachPoint);
+            }
+        }
+
+        /// <summary>
+        /// point where ToolTip connector is attached
+        /// </summary>
+        public Vector3 AnchorPosition
+        {
+            get
+            {
+                return anchor.transform.position;
+            }
+        }
+
+        /// <summary>
+        /// Tramsform of object to which ToolTip is attached
+        /// </summary>
+        public Transform ContentParentTransform
+        {
+            get
+            {
+                return contentParent.transform;
+            }
+        }
+
+        /// <summary>
+        /// is ToolTip active and displaying
+        /// </summary>
+        public bool IsOn
+        {
+            get
+            {
+                switch (masterTipState)
+                {
+                    case DisplayModeType.None:
+                    default:
+                        // Use our group state
+                        switch (groupTipState)
+                        {
+                            case DisplayModeType.None:
+                            default:
+                                // Use our local State
+                                switch (tipState)
+                                {
+                                    case DisplayModeType.None:
+                                    case DisplayModeType.Off:
+                                    default:
+                                        return false;
+
+                                    case DisplayModeType.On:
+                                        return true;
+
+                                    case DisplayModeType.OnFocus:
+                                        return HasFocus;
+                                }
+
+                            case DisplayModeType.On:
+                                return true;
+
+                            case DisplayModeType.Off:
+                                return false;
+
+                            case DisplayModeType.OnFocus:
+                                return HasFocus;
+                        }
+
+                    case DisplayModeType.On:
+                        return true;
+
+                    case DisplayModeType.Off:
+                        return false;
+
+                    case DisplayModeType.OnFocus:
+                        return HasFocus;
+                }
+            }
+        }
+
+        /// <summary>
+        /// does the ToolTip have focus.
+        /// </summary>
+        public bool HasFocus
+        {
+            get
+            {
+                return false;
+                //switch (ButtonState)
```

Comment:
is this going to be needed? if not, let's remove commented out code

Reply:
Addressed in my PR on his branch.